### ISSUE TITLE
System.Reflection.TargetParameterCountException: Parameter count mismatch. when using FooAuthenticationScheme to return AuthenticationResult.Success(ticket)

### ISSUE THREAD

**Issue by bachmannn** [ChilliCream/hotchocolate#3171]
**Describe the bug**

Exception gets thrown when adding a `FooAuthenticationScheme` to perform `HandleAuthenticateAsync()` method that `return AuthenticationResult.Success(ticket)`. 

Issue only replicable with HotChocolate requests and not regular MVC Controller.

This is added to `services.AddAuthentication()`, example below.

**Error**
```
fail: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware[1]
      An unhandled exception has occurred while executing the request.
      System.Reflection.TargetParameterCountException: Parameter count mismatch.
         at System.Reflection.RuntimeMethodInfo.InvokeArgumentsCheck(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
         at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
         at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
         at HotChocolate.Utilities.CombinedServiceProvider.Any(Type enumerableType, Object enumerable)
         at HotChocolate.Utilities.CombinedServiceProvider.Concat(Type enumerableType, Object enumerableA, Object enumerableB)
         at HotChocolate.Utilities.CombinedServiceProvider.GetService(Type serviceType)
         at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetService[T](IServiceProvider provider)
         at HotChocolate.SchemaBuilder.Setup.InitializeInterceptors[T](IServiceProvider services, IReadOnlyList`1 registered, List`1 interceptors)
         at HotChocolate.SchemaBuilder.Setup.CreateContext(SchemaBuilder builder, LazySchema lazySchema)
         at HotChocolate.SchemaBuilder.Setup.Create(SchemaBuilder builder)
         at HotChocolate.SchemaBuilder.Create()
         at HotChocolate.SchemaBuilder.HotChocolate.ISchemaBuilder.Create()
         at HotChocolate.Execution.RequestExecutorResolver.CreateSchemaAsync(NameString schemaName, RequestExecutorSetup options, IServiceProvider serviceProvider, CancellationToken cancellationToken)
         at HotChocolate.Execution.RequestExecutorResolver.CreateSchemaServicesAsync(NameString schemaName, RequestExecutorSetup options, CancellationToken cancellationToken)
         at HotChocolate.Execution.RequestExecutorResolver.GetRequestExecutorNoLockAsync(NameString schemaName, CancellationToken cancellationToken)
         at HotChocolate.Execution.RequestExecutorResolver.GetRequestExecutorAsync(NameString schemaName, CancellationToken cancellationToken)
         at HotChocolate.Execution.RequestExecutorProxy.GetRequestExecutorAsync(CancellationToken cancellationToken)
         at HotChocolate.AspNetCore.HttpPostMiddleware.HandleRequestAsync(HttpContext context, AllowedContentType contentType)
         at HotChocolate.AspNetCore.HttpPostMiddleware.InvokeAsync(HttpContext context)
         at HotChocolate.AspNetCore.WebSocketSubscriptionMiddleware.InvokeAsync(HttpContext context)
         at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
         at Microsoft.AspNetCore.Builder.Extensions.MapMiddleware.Invoke(HttpContext context)
         at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
         at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
         at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
info: Microsoft.AspNetCore.Hosting.Diagnostics[2]
```

**To Reproduce**

1. Add **FooAuthenticationScheme.cs**
```csharp
    public class FooAuthenticationScheme : AuthenticationHandler<AuthenticationSchemeOptions>
    {
        public const string Name = "Foo";

        public FooAuthenticationScheme(
            IOptionsMonitor<AuthenticationSchemeOptions> options,
            ILoggerFactory logger,
            UrlEncoder encoder,
            ISystemClock clock)
                : base(options, logger, encoder, clock)
        {
        }

        protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
        {
            bool isAnonymous = false;
            if (!isAnonymous)
            {
                var claims = new Claim[] { new Claim(ClaimTypes.Name, "FooUser") };
                var ticket = new AuthenticationTicket(new ClaimsPrincipal(new ClaimsIdentity(claims, Scheme.Name)), Scheme.Name);
           
                // Issue: exception thrown with HotChocolate Requests
                return await Task.FromResult(AuthenticateResult.Success(ticket));
            }

            // this works on HotChocolate (switch flag above)
            return await Task.FromResult(AuthenticateResult.NoResult());
        }
    }
}
```

2. Declare `FooAuthenticationScheme` in Startup.cs ConfigureServices method

```csharp
services.AddAuthentication("Foo")
     .AddScheme<AuthenticationSchemeOptions, FooAuthenticationScheme>
          (FooAuthenticationScheme.Name, FooAuthenticationScheme.Name, _ => { });

services.AddAuthorization();

services.AddGraphQLServer()
     .AddAuthorization()
     .AddQueryType(queryType => queryType.Name("Query"))
     .AddType<FooQuery>();
```

3. Because of Bug #2869 you have to put `.AllowAnonymous()` in `Configure` method which creates another issue by setting `Context.GetEndpoint()?.Metadata?.GetMetadata<IAllowAnonymous>() != null` true on all requests (will open seperate issue).

```csharp
app.UseEndpoints(endpoints =>
{
      endpoints.MapGraphQL()
            .AllowAnonymous();
}
```

**Expected behavior**
Request returned by `AuthenticateResult.Success(ticket)` is handled in the same way as Controller methods and exception is not thrown.

**Additional context**
`Version: 11.0.9`
`.NET 5.x`

**Comment by michaelstaib** [comment]
We will target these issues with the may update. This is actually a regression from 10 -> 11

**Comment by bachmannn** [comment]
> We will target these issues with the may update. This is actually a regression from 10 -> 11

After more testing I was able to consistently reproduce the bug when `DbContext` is used in `FooAuthenticationHandler`.

```csharp
        protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
        {
            var dbContext = Context.RequestServices.GetRequiredService<ApplicationDbContext>();

            var queryAnonymous = await dbContext.Foo.Select(bar => bar).ToListAsync();

            // Issue: exception thrown with HotChocolate Requests and DbContext used
            return await Task.FromResult(AuthenticateResult.Success(ticket));
        }
```

Not sure what in HC is causing it but I can look into it further, any guidance would be nice. From other issues it seems authentication is a good priority.

**Comment by bachmannn** [comment]
> We will target these issues with the may update. This is actually a regression from 10 -> 11

Is it going to be included in 11 still or the upcoming preview 12.0.0-preview-x?

Only seem to be getting application crashes now, unless I change HC package versions and run `dotnet restore`, it works temporarily, then crashes again. Custom AuthenticationSchemes are quite important for production applications.

What can I do to help you debug or find the root cause?

**Comment by bachmannn** [comment]
I found this issue is related to BCP. It sends a request token or has some caching that is causing HC to crash.

After `dotnet run`, if I refresh BCP before sending the request it works.
